### PR TITLE
refactor(wash-cli): deprecate wash app list in favor of wash app get

### PR DESCRIPTION
**Pull request by markkovari**
## Feature or Problem

Merging `wash app list` and `wash app get` commands.

I do need one clarification: we probably want to keep the periodically watched part of the list to the `wash app get` without a name, am I correct? If yes, I'll re-add it, and will write some tests. I just tested locally, but would bring more confidence if those would be automated. 

## Related Issues

fixes: #2884 

## Consumer Impact

Should reduce the number commands for the same thing.

### CODE REVIEW

**Review by vados-cosmonic — COMMENTED** [review]
Hey @markkovari this looks fantastic so far! Do we have this covered under tests properly? It might be nice to add a test case for the new syntax if we don't have one (I assume we have one for the non-list case of `wash get`.

**Comment by markkovari** [comment]
tl;dr

a, move the current `wash app list` functionality under the `wash app get` when no name
b, remove the wash app list and `wash app get` should return the yaml/json
c, something else

@vados-cosmonic 
no, I was about to check the tests and expected them to fail due to the removal of the other command. So I'll look at the test failures, but I still do not know if we just wanted to yoink all the yaml/json to the screen. Or we want to move the current `wash app list` render? Putting back to draft meanwhile

@brooksmtownsend you are the creator of the ticket can you (or anyone else) help me out?

**Comment by vados-cosmonic** [comment]
>  no, I was about to check the tests and expected them to fail due to the removal of the other command. So I'll look at the test failures, but I still do not know if we just wanted to yoink all the yaml/json to the screen. Or we want to move the current wash app list render? Putting back to draft meanwhile

I think moving the functionality around is definitely a good start -- if it just did exactly the same thing as before that'd be fine for a start I think!

**Comment by vados-cosmonic** [comment]
Rebasing to get the `oci`step fix in here so you can stop seeing red in the CI runs! :)

Thanks again for working on this

**Comment by markkovari** [comment]
Yes, I was kinda lost why are the golang tests failing if I change the command line parameters.
I will re-add then the `wash app list` and and rebase. (will come back to this tomorrow) Thanks all for the feedback.

File: crates/wash-cli/src/app/mod.rs [hunk 1]
```diff
@@ -125,14 +125,18 @@ pub struct PutCommand {
 
 #[derive(Args, Debug, Clone)]
 pub struct GetCommand {
-    /// The name of the application to retrieve
+    /// The name of the application to retrieve. If left empty retrieves all the applications, same as `wash app list`
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/app/mod.rs:128]: NIT: The first line will show up in rustdoc, so might be nice to leave a space after the first sentence here

File: crates/wash-cli/src/app/mod.rs [hunk 2]
```diff
@@ -125,14 +125,18 @@ pub struct PutCommand {
...
+    app_name: Option<String>,
 
     /// The version of the application to retrieve. If left empty, retrieves the latest version
     #[clap(name = "version")]
     version: Option<String>,
 
+    /// Enables Real-time updates, duration can be specified in ms or in humantime (eg: 5s, 2m, 15ms). Defaults to 1000 milliseconds.
+    #[clap(long,short, num_args = 0..=1, default_missing_value = "1000", value_parser = parse_watch_interval)]
+    pub watch: Option<std::time::Duration>,
+
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/app/mod.rs:139]: NIT: same note about the spacing in the comment

Also, wondering if we should default to milliseconds here or seconds since it's on human scale...

File: crates/wash-cli/src/app/mod.rs [hunk 3]
```diff
@@ -176,11 +180,16 @@ pub async fn handle_command(
...
+            get_application_list(cmd, &sp).await
         }
         Get(cmd) => {
-            sp.update_spinner_message("Getting application manifest ... ".to_string());
-            get_manifest(cmd).await
+            if let Some(app_name) = cmd.clone().app_name {
+                sp.update_spinner_message("Getting applications ... ".to_string());
+                get_manifest(cmd, &app_name).await
+            } else {
+                sp.update_spinner_message("Getting application manifest ... ".to_string());
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/app/mod.rs:190]:
```suggestion
            if let Some(app_name) = cmd.clone().app_name {
                sp.update_spinner_message("Getting applications... ".to_string());
                get_manifest(cmd, &app_name).await
            } else {
                sp.update_spinner_message("Getting application manifests... ".to_string());
```

File: crates/wash-cli/tests/common/mod.rs [hunk 4]
```diff
@@ -601,12 +601,12 @@ impl TestWashInstance {
             .context("failed to parse output of `wash app deploy`")
     }
 
-    /// Trigger the equivalent of `wash app list` on a [`TestWashInstance`]
-    pub(crate) async fn list_apps(&self) -> Result<AppListCommandOutput> {
+    /// Trigger the equivalent of `wash app get` on a [`TestWashInstance`]
+    pub(crate) async fn list_apps(&self) -> Result<AppGetCommandOutput> {
```
Comment (vados-cosmonic) [inline crates/wash-cli/tests/common/mod.rs:605]: Could we duplicate this test until we formally remove `wash app list` ?

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket: 

Nothing but nits!

File: crates/wash-cli/tests/common/mod.rs (hunk 4 above)
Comment (markkovari) [reply crates/wash-cli/tests/common/mod.rs:605]: Yes, I was thinking about as well. Re-added the two different version and tested the output of those. 🙇

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket: 

Thanks for tackling this @markkovari !
